KsApi/models/graphql/adapters/Project+ProjectFragment.swift

```diff
@@ -0,0 +1,119 @@
+      let dates = projectDates(from: projectFragment),
+      let locationFragment = projectFragment.location?.fragments.locationFragment,
+      let location = Location.location(from: locationFragment),
+      let photo = projectPhoto(from: projectFragment),
+      let state = projectState(from: projectFragment.state),
+      let userFragment = projectFragment.creator?.fragments.userFragment,
+      let creator = User.user(from: userFragment)
+    else { return nil }
+
+    let urls = Project.UrlsEnvelope(
+      web: UrlsEnvelope.WebEnvelope(project: projectFragment.url, updates: nil)
+    )
+
+    return Project(
+      blurb: projectFragment.description,
+      category: category,
+      country: country,
+      creator: creator,
+      memberData: MemberData(permissions: []),
+      dates: dates,
+      id: projectFragment.pid,
+      location: location,
+      name: projectFragment.name,
+      personalization: Personalization(),
+      photo: photo,
+      rewardData: RewardData(addOns: addOns, rewards: rewards),
+      slug: projectFragment.slug,
+      staffPick: projectFragment.isProjectWeLove,
+      state: state,
+      stats: projectStats(from: projectFragment),
+      urls: urls
+    )
+  }
+}
+
+private func projectRewards(from rewardFragments: [GraphAPI.RewardFragment]?) -> [Reward]? {
+  return rewardFragments?.compactMap { rewardFragment in
+    Reward.reward(from: rewardFragment)
+  }
+}
+
+/**
+ Returns a minimal `Project.Dates` from a `ProjectFragment`
+ */
+private func projectDates(from projectFragment: GraphAPI.ProjectFragment) -> Project.Dates? {
+  guard
+    let deadline = projectFragment.deadlineAt.flatMap(TimeInterval.init),
+    let launchedAt = projectFragment.launchedAt.flatMap(TimeInterval.init),
+    let stateChangedAt = TimeInterval(projectFragment.stateChangedAt)
+  else { return nil }
+
+  return Project.Dates(
+    deadline: deadline,
+    featuredAt: nil,
+    finalCollectionDate: finalCollectionDateTimeInterval(from: projectFragment.finalCollectionDate),
+    launchedAt: launchedAt,
+    stateChangedAt: stateChangedAt
+  )
+}
+
+private func finalCollectionDateTimeInterval(
+  from string: String?,
+  dateFormatter: ISO8601DateFormatter = ISO8601DateFormatter()
+) -> TimeInterval? {
+  guard let string = string else { return nil }
+
+  return dateFormatter.date(from: string)?.timeIntervalSince1970
+}
+
+/**
+ Returns a minimal `Project.Stats` from a `ProjectFragment`
+ */
+private func projectStats(from projectFragment: GraphAPI.ProjectFragment) -> Project.Stats {
+  return Project.Stats(
+    backersCount: projectFragment.backersCount,
+    commentsCount: nil,
+    convertedPledgedAmount: nil,
+    currency: projectFragment.currency.rawValue,
+    currentCurrency: nil,
+    currentCurrencyRate: nil,
+    goal: projectFragment.goal?.fragments.moneyFragment.amount.flatMap(Int.init) ?? 0,
+    pledged: projectFragment.pledged.fragments.moneyFragment.amount.flatMap(Int.init) ?? 0,
```

Comment:
are goal and pledged meant to be the same value?

Reply:
Nope, they aren't, they're just both coalesced to `0` as a fail-safe here.